tests/local/mod_accredible_evidenceitems_test.php

```diff
@@ -0,0 +1,367 @@
+
+        // Expect to call the endpoint once with url and reqdata.
+        $url = 'https://api.accredible.com/v1/credentials/1/evidence_items';
+        $questionsoutput = "<style>#main {  max-width: 780px;margin-left: auto;";
+        $questionsoutput .= "margin-right: auto;margin-top: 50px;margin-bottom: 80px; font-family: Arial;} ";
+        $questionsoutput .= "h1, h5 {   text-align: center;} ";
+        $questionsoutput .= ".answer { border: 1px solid grey; padding: 20px; font-size: 14px; ";
+        $questionsoutput .= "line-height: 22px; margin-bottom:30px; margin-top:30px;} ";
+        $questionsoutput .= "p {font-size: 14px; line-height: 18px;} </style>";
+        $questionsoutput .= "<div id='main'>";
+        $questionsoutput .= "<h1>" . $quiz->name . "</h1>";
+        $questionsoutput .= "<h5>Time Taken: 0 second</h5><div class='answer'></div></div>";
+        $evidenceitem = array(
+            "evidence_item" => array(
+                "description"   => $quiz->name,
+                "string_object" => $questionsoutput,
+                "hidden"        => true
+            )
+        );
+        $reqdata = json_encode($evidenceitem);
+
+        $mockclient1->expects($this->once())
+            ->method('post')
+            ->with($this->equalTo($url), $this->equalTo($reqdata))
+            ->willReturn($resdata);
+
+        $api = new apirest($mockclient1);
+        $evidenceitems = new evidenceitems($api);
+        $result = $evidenceitems->accredible_post_essay_answers($this->user->id, 1, 1);
+        $this->assertEquals($result, null);
+    }
+
+    /**
+     * Post credential evidence test
+     */
+    public function test_accredible_course_duration_evidence() {
+        global $DB;
+
+        $evidenceitems = new evidenceitems();
+
+        // When there are not enrolments.
+        $result = $evidenceitems->accredible_course_duration_evidence(1, 1, 1);
+        $this->assertEquals($result, null);
+
+        // When there is enrolment with timestart 0.
+        $enrolid = $this->create_enrolment(1);
+        $enrolid = $this->create_user_enrolment($enrolid, 2, 0);
+
+        $result = $evidenceitems->accredible_course_duration_evidence(2, 1, 1);
+        $this->assertEquals($result, null);
+
+        // When there is enrolment with timestart.
+        $enrolid = $this->create_enrolment(1);
+        $enrolid = $this->create_user_enrolment($enrolid, $this->user->id, time());
+
+        $mockclient1 = $this->getMockBuilder('client')
+            ->setMethods(['post'])
+            ->getMock();
+
+        // Mock API response data.
+        $resdata = $this->mockapi->resdata('evidence_items/create_success.json');
+
+        // Expect to call the endpoint once with url and reqdata.
+        $url = 'https://api.accredible.com/v1/credentials/1/evidence_items';
+
+        $mockclient1->expects($this->once())
+            ->method('post')
+            ->with($this->equalTo($url))
```

Comment:
I'd test it with the expected request body.

Reply:
Added